iOS/Flows/Conversation/Cells/MessagesTimeMachineCollectionViewLayout.swift

```diff
@@ -184,9 +184,14 @@ class MessagesTimeMachineCollectionViewLayout: TimeMachineCollectionViewLayout {
 
     /// If true, scroll to the most recent item after performing collection view updates.
     private var shouldScrollToEnd = false
+    private var insertedIndexPaths: Set<IndexPath> = []
     private var deletedIndexPaths: Set<IndexPath> = []
     /// How much to adjust the proposed scroll offset.
     private var scrollOffset: CGFloat = 0
+    /// The z position before update animations started
+    private var initialZPosition: CGFloat = 0
+    /// Items that that were visible before the animation started.
+    private var indexPathsVisibleBeforeAnimation: Set<IndexPath> = []
```

Comment:
Basically I kind of had to hack around the collection view layout behavior so I added these variables to help me do that.